### ISSUE TITLE
Reloading alert details page is fail

### ISSUE THREAD

**Issue by vykulakov** [alerta/alerta-webui#230]
**Issue Summary**
When I try to reload (or refresh, or press F5 or even Ctrl+F5) the page with alert details, I always see an empty page and a popup alert in my browser.

For example, I reload the page:http://192.168.1.25/alerta/alert/68f54758-f190-4557-8b84-bae6947518b4 and as a result, I see the following:
![image](https://user-images.githubusercontent.com/15159594/62517931-106d0800-b831-11e9-8bcd-17cc1a5baa93.png)
And after clicking on the OK button, I see an empty page.

In the web developer console, I see the following error:
```
Error: Network Error
    at t.exports (createError.js:16)
    at XMLHttpRequest.h.onerror (xhr.js:69)
localhost:8080/config:1 Failed to load resource: net::ERR_CONNECTION_REFUSED
createError.js:16 Uncaught (in promise) Error: Network Error
    at t.exports (createError.js:16)
    at XMLHttpRequest.h.onerror (xhr.js:69)
```
So Alerta WEB tries to use an incorrect Alerta endpoint (`localhost:8080` instead `192.168.1.25`). And as I can see, it happens because Alerta UI loads `config.json` from an incorrect place: http://192.168.1.25/alerta/alert/config.json instead http://192.168.1.25/alerta/config.json. So `config.json` is loaded with a relative URL.

This approach with the relative URL leads to errors in some cases so using `BASE_URL` or something should be more reliable.

**Environment**
- OS: Linux
- API version: 7.3.2
- Deployment: self-hosted
- WSGI environment: nginx/gunicorn

- Database: Postgres

- Server config:
  Auth enabled? Yes
  Auth provider? GitLab

- web UI version: 7.3.2
- CLI version: 7.3.0

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to an alert details page
2. Refresh the whole page (press F5)
4. See the above error

**Expected behavior**
I expect no errors in UI

**Additional context**
To solve the problem I use the following Nginx location to load `config.json`:
```
location ~ ^/alerta/(.+/)?config.json$ {
    alias /etc/alerta/config.json;
}
```
Though this solution is working now it cannot be considered as a good solution for all cases.

EDIT: Please, pay attention to prefix `/alerta` in my Nginx location. I use it because I run Alerta UI under the `/alerta` path.

**Comment by satterly** [comment]
Can you try this fix instead?
https://github.com/alerta/docker-alerta/commit/470a447a3573f8b505d53a0109fec3cd8c2e4658

**Comment by vykulakov** [comment]
Yeah, I've seen this MR and a related issue. And I use a similar solution as I described above.

But is it a final solution or maybe will you use BASE_PATH or something to point to `config.json`?

**Comment by satterly** [comment]
I could try base path but I have a feeling I tried that already and it didn't work for some reason. I'll try again.

**Comment by vykulakov** [comment]
OK, thank you.

**Comment by vykulakov** [comment]
I've found a very interesting (at least for me) approach to serve `config.json` in different environments (yes, it is a problem to provide yet another file for each environment):
```
location ~ ^/alerta/(.+/)?config.json$ {
    return 200 '{"endpoint": "$scheme://$host"}';
    add_header Content-Type application/json;
}
```
So it automatically uses the same protocol and hostname of a server as an endpoint for Alerta WEB UI and there is no need now to create this file manually or through ansible or something.

May be helpful in some environments.

**Comment by lucasalvatore1** [comment]
FYI having this same issue and none of the methods mentioned here are working for me.
I see this issue when trying to refresh on the alerts page, or when going directly to an alert URL e.g. http://alerta/alert/94ff7303-bf70-49d8-85bf-43f439636d00


**Environment**

- Ubuntu 18.04
- Alerta version 7.3.1
- Deployment: self-hosted
- WSGI environment: nginx/gunicorn
- Database: Postgres
- Server config:
- Auth enabled? Yes

Current nginix config:

```
(alerta) root@alerta-dev:/etc/nginx/sites-enabled# cat default
server {

        listen 80 default_server deferred;

        access_log /var/log/nginx/access.log;
	error_log /var/log/nginx/error.log notice;

        location /api/ {
                proxy_pass http://backend/;
                proxy_set_header Host $host:$server_port;
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
		error_page 405   =200 $uri;
        }

        location / {
		try_files $uri $uri/ /index.html?/$request_uri;
                root /var/www/html/;
                error_page 405   =200 $uri;
      }
 	location ~ ^/alerta/(.+/)?config.json$ {
    		alias /var/www/html/config.json;
	}
}

upstream backend {
        server localhost:8080 fail_timeout=0;
}
```

Error in java script dev console:

```
GET http://localhost:8080/config net::ERR_CONNECTION_REFUSED
Uncaught (in promise) Error: Network Error
    at t.exports (createError.js:16)
    at XMLHttpRequest.h.onerror (xhr.js:69)
```

**Comment by vykulakov** [comment]
@lucasalvatore1 My code above is based on an idea that Alerta UI is available under the `/alerta` path prefix. But as I can see in your setup Alerta UI is available under the root path (I mean this location: `location / {`). So you must use the location like this `location ~ ^/(.+/)?config.json$ {` for `config.json` (without any additional prefixes like `/alerta`).

**Comment by satterly** [comment]
And I'm not sure that "$request_uri" is necessary, or could even be a problem.
```
  try_files $uri $uri/ /index.html?/$request_uri;
```
Should just be ...
```
  try_files $uri $uri/ /index.html;
```
See https://github.com/alerta/alerta-webui#html5-history-mode and https://router.vuejs.org/guide/essentials/history-mode.html#example-server-configurations

**Comment by satterly** [comment]
@vykulakov Are you able to review and/or test this PR #232 ? Thanks.

**Comment by lucasalvatore1** [comment]
@satterly i tried your suggestion and it didn't change anything.
However the suggestion from @vykulakov seems to have fixed it.  This is my nginx config which is working now:

```
server {

        listen 80 default_server deferred;

        access_log /var/log/nginx/access.log;
	error_log /var/log/nginx/error.log notice;

        location /api/ {
                proxy_pass http://backend/;
                proxy_set_header Host $host:$server_port;
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
		error_page 405   =200 $uri;
        }
	location ~ ^/(.+/)?config.json$ {
		alias /var/www/html/config.json;
	}
        location / {
		try_files $uri $uri/ /index.html?/$request_uri;
                root /var/www/html/;
                error_page 405   =200 $uri;
      }
}
upstream backend {
        server localhost:8080 fail_timeout=0;
}
```

**Comment by vykulakov** [comment]
@satterly Left a couple of comments on your PR.

**Comment by satterly** [comment]
@lucasalvatore1 with the fix in #232 you shouldn't need this hack anymore ...

```
	location ~ ^/(.+/)?config.json$ {
		alias /var/www/html/config.json;
	}
```
